src/lib/Components/Bidding/Screens/ConfirmBid.tsx

```diff
@@ -30,12 +33,121 @@ interface ConfirmBidProps extends ViewProperties {
+  mutation ConfirmBidMutation($input: BidderPositionInput!) {
+    createBidderPosition(input: $input) {
+      result {
+        position {
+          id
+        }
+        status
+        message_header
+        message_description
+        message_description_md
+      }
+    }
+  }
+`
+
+export class ConfirmBid extends React.Component<ConfirmBidProps, ConformBidState> {
+  state = {
+    pollCount: 0,
+    intervalToken: 0,
+  }
+
   onPressConditionsOfSale = () => {
     SwitchBoard.presentModalViewController(this, "/conditions-of-sale?present_modally=true")
   }
+
+  placeBid() {
+    commitMutation(this.props.relay.environment, {
+      onCompleted: (results, errors) => {
+        this.verifyBidPosition(results, errors)
+      },
+      onError: e => {
+        // TODO catch error!
+        // this.verifyAndShowBidResult(null, e)
+        console.log("error!", e, e.message)
+      },
+      mutation: bidderPositionMutation,
+      variables: {
+        input: {
+          sale_id: this.props.sale_artwork.sale.id,
+          artwork_id: this.props.sale_artwork.artwork.id,
+          max_bid_amount_cents: this.props.bid.cents,
+        },
+      },
+    })
+  }
+
+  verifyBidPosition(results, errors) {
+    const status = results.createBidderPosition.result.status
+    if (!errors && status === "SUCCESS") {
+      const positionId = results.createBidderPosition.result.position.id
+      const query = `
+        {
+          me {
+            bidder_position(id: "${positionId}") {
+              processed_at
+              is_active
+            }
+          }
+        }
+      `
+      const interval = setInterval(() => {
+        metaphysics({ query }).then(this.checkBidPosition.bind(this))
+      }, 2000)
+      this.setState({ intervalToken: interval })
+    } else {
+      const message_header = results.createBidderPosition.result.message_header
+      const message_description_md = results.createBidderPosition.result.message_description_md
+      this.showBidResult(false, message_header, message_description_md)
+    }
+  }
+
+  checkBidPosition(result) {
+    const bidderPosition = result.data.me.bidder_position
+    if (bidderPosition.processed_at) {
+      clearInterval(this.state.intervalToken)
+      if (bidderPosition.is_active) {
+        // wining
+        this.showBidResult(true)
+      } else {
+        // outbid
+        this.showBidResult(false)
+      }
+    } else {
+      // poll again
+      if (this.state.pollCount > MAX_POLL_ATTEMPTS) {
+        clearInterval(this.state.intervalToken)
+      }
+      this.setState({ pollCount: this.state.pollCount + 1 })
+    }
+  }
+
+  showBidResult(winning: boolean, messageHeader?: string, messageDescriptionMd?: string) {
+    this.props.navigator.push({
+      component: BidResult,
+      title: "",
+      passProps: {
+        message_header: messageHeader,
+        message_description_md: messageDescriptionMd,
+        winning,
+      },
+    })
+  }
```

Comment:
Should we move the bid polling logic into its own file, so we can extensively unit test it? Or is it common in React to have components do networking stuff like this?

In either case, we should add some tests for the polling code. Maybe just a `#TODO` for now?

Reply:
Thats a good idea. I will put a TODO note and do that in a follow up PR.